exporter/awsemfexporter/metric_translator.go

```diff
@@ -0,0 +1,296 @@
+// limitations under the License.
+
+package awsemfexporter
+
+import (
+	"bytes"
+	"crypto/sha1"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"time"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter/mapwithexpiry"
+)
+
+const (
+	CleanInterval = 5 * time.Minute
+	MinTimeDiff   = 50 // We assume 50 micro-seconds is the minimal gap between two collected data sample to be valid to calculate delta
+
+	//The following constants indicate the existence of resource attribute service.name and service.namespace
+	ServiceNameOnly = iota
+	ServiceNamespaceOnly
+	ServiceNameAndNamespace
+	ServiceNotDefined
+
+	OtlibDimensionKey            = "OTLib"
+	defaultNameSpace             = "default"
+	noInstrumentationLibraryName = "Undefined"
+
+	// See: http://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html
+	maximumLogEventsPerPut = 10000
+)
+
+var currentState = mapwithexpiry.NewMapWithExpiry(CleanInterval)
+
+type rateState struct {
+	value     interface{}
+	timestamp int64
+}
+
+// CWMetrics defines
+type CWMetrics struct {
+	Measurements []CwMeasurement
+	Timestamp    int64
+	Fields       map[string]interface{}
+}
+
+// CwMeasurement defines
+type CwMeasurement struct {
+	Namespace  string
+	Dimensions [][]string
+	Metrics    []map[string]string
+}
+
+// CWMetric stats defines
+type CWMetricStats struct {
+	Max   float64
+	Min   float64
+	Count *wrappers.Int64Value
+	Sum   *wrappers.DoubleValue
+}
+
+// TranslateOtToCWMetric converts OT metrics to CloudWatch Metric format
+func TranslateOtToCWMetric(md *consumerdata.MetricsData) ([]*CWMetrics, int) {
+	var cwMetricLists []*CWMetrics
+	namespace := defaultNameSpace
+	svcAttrMode := ServiceNotDefined
+	totalDroppedMetrics := 0
+
+	if attributes := md.Resource.GetLabels(); attributes != nil {
+		serviceName, svcNameOk := attributes[conventions.AttributeServiceName]
+		serviceNamespace, svcNsOk := attributes[conventions.AttributeServiceNamespace]
+
+		if svcNameOk {
+			svcAttrMode = ServiceNameOnly
+		}
+		if svcNsOk {
+			if ServiceNameOnly == svcAttrMode {
+				svcAttrMode = ServiceNameAndNamespace
+			} else {
+				svcAttrMode = ServiceNamespaceOnly
+			}
+		}
+		switch svcAttrMode {
+		case ServiceNameAndNamespace:
+			namespace = fmt.Sprintf("%s/%s", serviceNamespace, serviceName)
+		case ServiceNameOnly:
+			namespace = serviceName
+		case ServiceNamespaceOnly:
+			namespace = serviceNamespace
+		case ServiceNotDefined:
+		default:
+		}
+	}
+	for _, metric := range md.Metrics {
+		if metric == nil || metric.MetricDescriptor == nil {
+			totalDroppedMetrics += len(metric.GetTimeseries())
+			continue
+		}
+		//TODO: Handle OTLib as a dimension when it's supported
```

Comment:
I think we're missing this since we converted to OC, but we should be sticking to the native OTel format